build.wake

```diff
@@ -0,0 +1,73 @@
+tuple DevicetreeOverlayGeneratorOptions =
+  global TopDTSFile:    Path
+  global OtherDTSFiles: List Path
+  global Type:          String
+  global OutputFile:    String
+
+# Instead of copying the script sources into build/{here}, we'll just
+# execute directly out of the package directory, since nothing gets
+# modified during execution.
+#
+# The virtualenv does get placed in build/{here} by
+# addPythonRequirementsEnv, along with a copy of requirements.txt
+def generatorDir = here
+
+#######################################################################
+# makeDevicetreeOverlayGeneratorOptions takes the following parameters:
+#   - topDTSFile: The top-level Devicetree source file
+#   - otherDTSFiles: any other Devicetree source files included in the
+#                    heirarchy of Devicetree source files
+#   - type: One of {rtl, arty, qemu, hifive, spike, vc707}
+#   - outputFile: A string representing the path of the output file to
+#                 produce
+#######################################################################
+global def makeDevicetreeOverlayGeneratorOptions topDTSFile otherDTSFiles type outputFile =
+  DevicetreeOverlayGeneratorOptions topDTSFile otherDTSFiles type outputFile
+
+global def runDevicetreeOverlayGenerator options =
+  def topDTSFile = options.getDevicetreeOverlayGeneratorOptionsTopDTSFile
+  def otherDTSFiles = options.getDevicetreeOverlayGeneratorOptionsOtherDTSFiles
+
+  def inputs =
+    def generatorSources = sources here `.*\.py`
+    def dtsSources = topDTSFile, otherDTSFiles
+    generatorSources ++ dtsSources
+
+  def outputs = options.getDevicetreeOverlayGeneratorOptionsOutputFile, Nil
+
+  def args =
+    def output =
+      "-o", options.getDevicetreeOverlayGeneratorOptionsOutputFile,
+      Nil
+
+    def type =
+      "-t", options.getDevicetreeOverlayGeneratorOptionsType,
+      Nil
+
+    # The generated overlay will include an `/include/` directive at the top of
+    # the file which points at the topDTSFile. To make sure that that directive
+    # includes the filename only and not the fully-qualified path, we can pass
+    # the --rename-include option.
+    def renameInclude =
+      def notdir str =
+        match (matches `.*/.*` str)
+          True  = extract `.*/(.*)` str | head | getOrElse ""
+          False = str
```

Comment:
Since your `extract` expression in the middle already accounts for optionality, there's no need to do both that and have the outer `match` statement. How about rewriting the entire `match` statement as:

```wake
extract `.*/(.*)` str | head | getOrElse str
```